contracts/mainnet/connectors/morpho-compound/main.sol

```diff
@@ -0,0 +1,440 @@
+
+			(, , uint256 _amtDebt) = morphoCompoundLens
+				.getCurrentBorrowBalanceInOf(_poolTokenAddress, address(this));
+
+			_amt = _amtDSA <= _amtDebt ? _amtDSA : _amtDebt;
+		}
+
+		if (_isETH) convertEthToWeth(_isETH, _tokenContract, _amt);
+
+		approve(_tokenContract, address(morphoCompound), _amt);
+
+		morphoCompound.repay(_poolTokenAddress, address(this), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogPayback(address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_amt,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Payback ETH/ERC20_Token.
+	 * @notice Payback a token to Morpho Compound on behalf of a user.
+	 * @param _tokenAddress The address of underlying token to payback.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to payback.(For ETH: cWETH address)
+	 * @param _onBehalf The address of user who's debt to repay.
+	 * @param _amount The amount of the token (in underlying) to payback. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens paid back.
+	 */
+	function paybackOnBehalf(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		address _onBehalf,
+		uint256 _amount,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		bool _isETH = _tokenAddress == ethAddr;
+		uint256 _amt = getUint(_getId, _amount);
+		
+		TokenInterface _tokenContract = _isETH 
+			? TokenInterface(wethAddr) 
+			: TokenInterface(_tokenAddress);
+
+		if (_amt == uint256(-1)) {
+			uint256 _amtDSA = _isETH
+				? address(this).balance
+				: _tokenContract.balanceOf(address(this));
+
+			(, , uint256 _amtDebt) = morphoCompoundLens
+				.getCurrentBorrowBalanceInOf(_poolTokenAddress, _onBehalf);
+
+			_amt = _amtDSA <= _amtDebt ? _amtDSA : _amtDebt;
+		}
+
+		if (_isETH) convertEthToWeth(_isETH, _tokenContract, _amt);
+
+		approve(_tokenContract, address(morphoCompound), _amt);
+
+		morphoCompound.repay(_poolTokenAddress, _onBehalf, _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogPaybackOnBehalf(address,address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_onBehalf,
+			_amt,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Claim rewards.
+	 * @notice Claim rewards for the given assets from underlying protocol.
+	 * @param _poolTokenAddresses The cToken addresses to claim rewards from..(For ETH: cToken address of WETH)
+	 * @param _tradeForMorphoToken Whether or not to trade COMP tokens for MORPHO tokens.
+	 */
+	function claim(
+		address[] calldata _poolTokenAddresses,
+		bool _tradeForMorphoToken
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		morphoCompound.claimRewards(_poolTokenAddresses, _tradeForMorphoToken);
```

Comment:
Same as in Morpho Aave an reward amount is returned. Since Compound is emitting COMP rewards, idk if it might be helpful to know how many COMP rewards you claimed 🙂

Reply:
Yes, Would be good for logging! Added.